test/unit/eval_algorithms/test_answer_relevance.py

```diff
@@ -0,0 +1,190 @@
+        generated_questions: str
+
+    @pytest.mark.parametrize(
+        "test_case",
+        [
+            #
+            TestCaseAnswerRelevanceEvaluate(
+                input_dataset=DATASET_WITH_CONTEXT,
+                dataset_config=DataConfig(
+                    dataset_name="my_custom_dataset",
+                    dataset_uri="tba",
+                    dataset_mime_type=MIME_TYPE_JSON,
+                    model_input_location="tba",
+                    target_output_location="tba",
+                    model_output_location=None,
+                    category_location="tba",
+                ),
+                question_vector=QUESTION_EMBEDDINGS,
+                gen_question_vector=GEN_QUESTION_EMBEDDINGS,
+                generated_questions="Question: What was the date of the first Super Bowl?",
+                expected_response=[
+                    EvalOutput(
+                        eval_name="answer_relevance",
+                        dataset_name="my_custom_dataset",
+                        prompt_template=None,
+                        dataset_scores=DATASET_SCORES,
+                        output_path="/tmp/eval_results/answer_relevance_my_custom_dataset.jsonl",
+                    )
+                ],
+            ),
+        ],
+    )
+    @patch("fmeval.eval_algorithms.answer_relevance.get_dataset")
+    def test_answer_relevance_evaluate(self, mock_get_dataset, test_case):
+        """
+        GIVEN datasets.
+        WHEN the Answer Relevance evaluate method is called.
+        THEN the correct EvalOutput is returned.
+
+        Note: this is basically an integration test rather than a unit test like `test_evaluate` above.
+        It uses a special toy dataset where we are able to compute the exact expected scores by hand.
+        The purpose of this test is really to ensure that the correct scores are being generated.
+        """
+        # GIVEN
+        mock_judge_model_runner = Mock()
+        mock_judge_model_runner.predict.side_effect = [
+            (test_case.generated_questions, None),
+        ]
+        mock_embedding_model_runner = Mock()
+        mock_embedding_model_runner.predict.side_effect = [test_case.question_vector, test_case.gen_question_vector]
+        mock_get_dataset.return_value = test_case.input_dataset
+        eval_algorithm = AnswerRelevance()
+        # WHEN
+        actual_response = eval_algorithm.evaluate(
+            judge_model=mock_judge_model_runner,
+            embeddings_model=mock_embedding_model_runner,
+            dataset_config=test_case.dataset_config,
+            save=True,
+        )
+        # THEN
+        assert actual_response == test_case.expected_response
+
+    class TestCaseAnswerRelevanceScore(NamedTuple):
+        question: str
+        gen_questions_str: str
+        question_vector: List[float]
+        gen_question_vector: List[float]
+        expected_score: float
+
+    @pytest.mark.parametrize(
```

Comment:
Could you please add a test case where there are multiple generated question vectors? I know the computation of the mean is quite trivial, but right now, we technically have no test that is validating the mean computation.